### ISSUE TITLE
Change "TLS" to "TLS, If Available" or "STARTTLS"

### ISSUE THREAD

**Issue by Helithumper** [CTFd/CTFd#2292]
When configuring CTFd for SunshineCTF 2022, we had issues due to an email misconfiguration.

In order to ensure that other administrators do not run into confusion, I propose changing the "SSL" and "TLS" options in the email config to "SSL" and "TLS, If Available" or "STARTTLS" in order to ensure that the confusion isn't there for users who don't know SMTP that well (myself included).

<img width="575" alt="image" src="https://user-images.githubusercontent.com/11335266/233904889-ebbaa53f-b88d-4321-afe7-5af414dd22e3.png">

Initially I proposed making these radio buttons as I had confused [SMTP SSL/TLS for HTTPS SSL/TLS](https://www.fastmail.help/hc/en-us/articles/360058753834-SSL-TLS-and-STARTTLS) but as I later learned the two binary options are "Is this connection using SSL or not" and "Has a STARTTLS command been sent on the connection".

This is apparently a known confusion that even the Thunderbird client has run into in the past and their solution was the following:
<img width="339" alt="image" src="https://user-images.githubusercontent.com/11335266/233905209-7fb59efd-394f-4e8a-91b4-eae7f23225a4.png">

TLDR: I think that the "TLS" option should be replaced with either "TLS, if Available" or "STARTTLS" and "SSL" replaced with "SSL/TLS".

**Comment by Helithumper** [comment]
Mainly asking for feedback on which direction, I'll also make a quick PR for changing the options to the Thunderbird way in the meantime and we can modify as needed

**Comment by Helithumper** [comment]
That should clear up confusion around the two configuration settings. Part of me wants to add a link to some documentation somewhere on the differences between the two as it was very easy to screw up however this is a good start I think

**Comment by ColdHeat** [comment]
Yes I could write out some documentation and then link to it in the SMTP settings panel. Is there anything specific that you want to highlight that helped you figure this out?

**Comment by Helithumper** [comment]
[This guide](https://www.fastmail.help/hc/en-us/articles/360058753834-SSL-TLS-and-STARTTLS) was probably the most useful guide out of all of them as it's relatively simple. 

The first paragraph of that article is the core of it.
>[SSL and TLS](https://en.wikipedia.org/wiki/Transport_Layer_Security) are the standard >technology to encrypt connections between two computers. This prevents any third parties from >spying on these communications.

>TLS is the successor to SSL. It is supported by all modern and secure systems that handle >internet traffic, including Fastmail. The terms SSL and TLS are often switched and used >interchangeably.

>[STARTTLS](https://en.wikipedia.org/wiki/Starttls) is different to SSL and TLS. Before >encryption was standard, many connections between an email client and the server were done >insecurely. This put personal information in danger of being stolen. STARTTLS helped to reduce >this risk by taking an existing insecure connection and upgrading it to a secure connection >that used SSL/TLS. STARTTLS works with either SSL or TLS.

and the section on "TLS vs STARTTLS naming problem" sums up the change.

>TLS vs STARTTLS naming problem

> One cause of confusion around the names of these different technologies is that some email software incorrectly uses the term TLS when they should have used STARTTLS.
> 
> Older versions of Thunderbird in particular used "TLS" to mean that STARTTLS should be used to upgrade the connection, and the connection should fail if STARTTLS is not supported.
> 
> These versions also used the term "TLS, if available". "TLS, if available" meant that the program would try to use STARTTLS to upgrade the connection if the server supported this, but otherwise use an insecure connection.
>

**Comment by Helithumper** [comment]
> email_settings = {
> 
>     "SMTP_HOST": "smtp.example.com",  # Your SMTP service
> 
>     "SMTP_PORT": 587,  # SMTP service port (usually 587 for TLS or 465 for SSL)
> 
>     "USE_SSL/TLS": "SSL/TLS",  # Choose SSL/TLS or try TLS if available (as described)
> 
>     "USERNAME": "your_username",  # Your username for SMTP server authentication
> 
>     "PASSWORD": "your_password",  # Your password for SMTP server authentication
> 
>     "DEFAULT_RECIPIENT": "default_recipient@example.com"  # Default recipient address
> 
> }
> 
> 

What is this in reference to? Are you pointing out that there's a config somewhere that still references the old naming scheme?
